pkg/client/opensergo_client_observer.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2022, OpenSergo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"fmt"
+	"github.com/golang/groupcache"
+	"github.com/opensergo/opensergo-go/pkg/common/logging"
+	"github.com/opensergo/opensergo-go/pkg/configkind"
+	transportPb "github.com/opensergo/opensergo-go/pkg/proto/transport/v1"
+	"github.com/opensergo/opensergo-go/pkg/transport"
+	"github.com/opensergo/opensergo-go/pkg/transport/subscribe"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/anypb"
+	"reflect"
+)
+
+// SubscribeConfigStreamObserver is a struct as an observer role to listen the response of SubscribeConfig() from opensergo-control-plane.
+type SubscribeConfigStreamObserver struct {
+	openSergoClient *OpenSergoClient
+
+	subscribeResponseChan  chan transportPb.SubscribeResponse
```

Comment:
Maybe we should carry pointers rather than values (to avoid copy)?

Reply:
optimized.